src/hooks/connectWeb3/useActiveWeb3React.tsx

```diff
@@ -21,12 +25,48 @@ type Web3ContextType = {
 const [EMPTY_CONNECTOR, EMPTY_HOOKS] = initializeConnector<Connector>(() => EMPTY)
 const EMPTY_STATE = { connector: EMPTY_CONNECTOR, hooks: EMPTY_HOOKS }
 const EMPTY_CONTEXT: Web3ContextType = { connector: EMPTY }
-const Web3Context = createContext(EMPTY_CONTEXT)
+export const Web3Context = createContext(EMPTY_CONTEXT)
 
 export default function useActiveWeb3React() {
   return useContext(Web3Context)
 }
 
+export function getNetwork(jsonRpcEndpoint?: string | JsonRpcProvider) {
+  if (jsonRpcEndpoint) {
+    let connector, hooks
+    if (JsonRpcProvider.isProvider(jsonRpcEndpoint)) {
+      ;[connector, hooks] = initializeConnector((actions) => new JsonRpcConnector(actions, jsonRpcEndpoint))
+    } else {
+      ;[connector, hooks] = initializeConnector((actions) => new Url(actions, jsonRpcEndpoint))
+    }
+    connector.activate()
+    return { connector, hooks }
+  }
+  return EMPTY_STATE
+}
+
+function getWallet(provider?: JsonRpcProvider | Eip1193Provider) {
+  if (provider) {
+    let connector, hooks
+    if (JsonRpcProvider.isProvider(provider)) {
+      ;[connector, hooks] = initializeConnector((actions) => new JsonRpcConnector(actions, provider))
+    } else if (JsonRpcProvider.isProvider((provider as any).provider)) {
+      throw new Error('Eip1193Bridge is experimental: pass your ethers Provider directly')
+    } else {
+      ;[connector, hooks] = initializeConnector((actions) => new EIP1193(actions, provider))
+    }
+    connector.activate()
+    return { connector, hooks }
+  }
+  return EMPTY_STATE
+}
+
+export function useActiveProvider(): Web3Provider | undefined {
+  const activeWalletProvider = getPriorityConnector(...connections).usePriorityProvider() as Web3Provider
+  const { connector: network } = getNetwork() // Return network-only provider if no wallet is connected
```

Comment:
is this always going to return the empty connector?

Reply:
If the integrator provides a JSON RPC URL, it'll return a network-only connector (read-only, to get price quotes)
Else, it'll be the empty connector.


(once we have the RPC URL fallback implemented, it should always return a network connector)